addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3215,11 +3231,53 @@ options.registry.MegaMenuLayout = options.registry.SelectTemplate.extend({
     _computeWidgetState: function (methodName, params) {
         if (methodName === 'selectTemplate') {
             const templateDefiningClass = this.containerEl.querySelector('section')
-                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu'))[0];
+                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu_'))[0];
             return `website.${templateDefiningClass}`;
         }
         return this._super(...arguments);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Puts back original template.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    async _onResetTemplate(ev) {
+        const widgetValue = this._computeWidgetState('selectTemplate');
+        const template = await this._rpc({
+            model: 'ir.ui.view',
+            method: 'render_public_asset',
+            args: [widgetValue, {}],
+        });
+        this.containerEl.insertAdjacentHTML('beforeend', template);
+    },
```

Comment:
Try to avoid remaking the RPC if not necessary, the SelectTemplate option has a cache.

Reply:
This cache is available only when the dropdown has been opened by the user. But yes I can check th ecache before making a RPC

Reply:
My point goes both ways: you should also not update the cache on widget opening if the rpc was already done on template reset. And there is no need of code to handle that: you will **reduce** the code by implementing it properly:
- Dedicated function to do the RPC and update the cache
- Call that function on widget opening and on template reset

=> No duplicated code, les code and more efficient.

Reply:
Ok I see! Can't we also get rid of `_onWidgetOpening` and retrieve the template (from cache or through rpc) directly in selectTemplate ? 
![Screenshot from 2021-09-14 09-45-04](https://user-images.githubusercontent.com/49784121/133216826-1544ee85-331a-4140-997c-aad6783c8903.png)
![Screenshot from 2021-09-14 09-44-56](https://user-images.githubusercontent.com/49784121/133216830-f295957d-b8df-408c-b65e-93044cab43ef.png)

Reply:
If you do that, you will have to wait for a RPC each first time you hover an option in that select which will thus create a flickering.